### ISSUE TITLE
Bench easy install error: subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2

### ISSUE THREAD

**Issue by yashodhank** [frappe/bench#838]
Fresh Ubuntu 18.04 VPS with 1GB RAM

same result with xyz, frappe sudo enabled users and later with root
all attempts of installing failed with same error. 

```
lZ_mysql_user.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1567279072.6-176384184708260/AnsiballZ_mysql_user.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1567279072.6-176384184708260/ > /dev/null 2>&1 && sleep 0'
failed: [localhost] (item=::1) => {
    "ansible_loop_var": "item", 
    "changed": false, 
    "invocation": {
        "module_args": {
            "append_privs": false, 
            "ca_cert": null, 
            "check_implicit_admin": true, 
            "client_cert": null, 
            "client_key": null, 
            "config_file": "/root/.my.cnf", 
            "connect_timeout": 30, 
            "encrypted": false, 
            "host": "::1", 
            "host_all": false, 
            "login_host": "localhost", 
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "login_port": 3306, 
            "login_unix_socket": null, 
            "login_user": null, 
            "name": "root", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "priv": null, 
            "sql_log_bin": true, 
            "state": "present", 
            "update_password": "always", 
            "user": "root"
        }
    }, 
    "item": "::1", 
    "msg": "The PyMySQL (Python 2.7 and Python 3.X) or MySQL-python (Python 2.X) module is required."
}

RUNNING HANDLER [mariadb : restart mysql] *****************************************************************************************************************************
task path: /tmp/.bench/playbooks/roles/mariadb/handlers/main.yml:2

PLAY RECAP ************************************************************************************************************************************************************
localhost                  : ok=22   changed=5    unreachable=0    failed=1    skipped=15   rescued=0    ignored=0   

Traceback (most recent call last):
  File "install.py", line 413, in <module>
    install_bench(args)
  File "install.py", line 135, in install_bench
    run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
  File "install.py", line 327, in run_playbook
    success = subprocess.check_call(args, cwd=os.path.join(cwd, 'playbooks'))
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2
```

**Comment by yashodhank** [comment]
Ok, found the culprit.
redis-server config
```
- bind 127.0.0.1 ::1
+ bind 127.0.0.1
```
removing or commenting `bind 127.0.0.1 ::1` from redis config file on `/etc/redis/redis.conf` fixed the issue on my ubuntu 18.04 VPS.

This VPS does not have IPv6 and currently disabled it through `/etc/sysctl.conf`

There is already redis template but it seems to not taking effect and bench install fails due to IPv6 non-availability.

**Comment by Nofal-Mami** [comment]
Hi,
@yashodhank I am facing same issue, I did what you mentioned as solution, no luck still issue showing.

I am using ubuntu WSL under windows 10.

PLAY RECAP *************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

Traceback (most recent call last):
  File "install.py", line 413, in <module>
    install_bench(args)
  File "install.py", line 135, in install_bench
    run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
  File "install.py", line 327, in run_playbook
    success = subprocess.check_call(args, cwd=os.path.join(cwd, 'playbooks'))
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2

**Comment by yashodhank** [comment]
@Nofal-Mami Please open new discussion thread at https://discuss.erpnext.com

**Comment by Nofal-Mami** [comment]
already done, thx

https://discuss.erpnext.com/t/how-to-install-erpnext-on-ubuntu-18-04-wsl/52542

**Comment by yashodhank** [comment]
`Bench` failed on another fresh VPS with latest updates installed..

```
TASK [redis : Install apt packages] *************************************************************************************************************************************************
task path: /tmp/.bench/playbooks/roles/redis/tasks/main.yml:9
[DEPRECATION WARNING]: Invoking "apt" only once while using a loop via squash_actions is deprecated. Instead of using a loop to supply multiple items and specifying `pkg: "{{ item
}}"`, please use `pkg: ['redis-server']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~root && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131 `" && echo ansible-tmp-1575690018.0715492-231361916483131="` echo /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131 `" ) && sleep 0'
Using module file /usr/local/lib/python3.6/dist-packages/ansible/modules/packaging/os/apt.py
<127.0.0.1> PUT /home/frappe/.ansible/tmp/ansible-local-8189noxi_06e/tmpw06q39hb TO /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131/AnsiballZ_apt.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131/ /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131/AnsiballZ_apt.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131/AnsiballZ_apt.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1575690018.0715492-231361916483131/ > /dev/null 2>&1 && sleep 0'
failed: [localhost] (item=['redis-server']) => {
    "ansible_loop_var": "item",
    "cache_update_time": 1575689975,
    "cache_updated": false,
    "changed": false,
    "invocation": {
        "module_args": {
            "allow_unauthenticated": false,
            "autoclean": false,
            "autoremove": false,
            "cache_valid_time": 0,
            "deb": null,
            "default_release": null,
            "dpkg_options": "force-confdef,force-confold",
            "force": true,
            "force_apt_get": false,
            "install_recommends": null,
            "name": [
                "redis-server"
            ],
            "only_upgrade": false,
            "package": [
                "redis-server"
            ],
            "policy_rc_d": null,
            "purge": false,
            "state": "present",
            "update_cache": null,
            "upgrade": null
        }
    },
    "item": [
        "redis-server"
    ],
    "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"   --force-yes   install 'redis-server'' failed: W: --force-yes is deprecated, use one of the options starting with --allow instead.\nE: Sub-process /usr/bin/dpkg returned an error code (1)\n",
    "rc": 100,
    "stderr": "W: --force-yes is deprecated, use one of the options starting with --allow instead.\nE: Sub-process /usr/bin/dpkg returned an error code (1)\n",
    "stderr_lines": [
        "W: --force-yes is deprecated, use one of the options starting with --allow instead.",
        "E: Sub-process /usr/bin/dpkg returned an error code (1)"
    ],
    "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nThe following additional packages will be installed:\n  libjemalloc1 redis-tools\nSuggested packages:\n  ruby-redis\nThe following NEW packages will be installed:\n  libjemalloc1 redis-server redis-tools\n0 upgraded, 3 newly installed, 0 to remove and 1 not upgraded.\nNeed to get 634 kB of archives.\nAfter this operation, 3012 kB of additional disk space will be used.\nGet:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjemalloc1 amd64 3.6.0-11 [82.4 kB]\nGet:2 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 redis-tools amd64 5:4.0.9-1ubuntu0.2 [516 kB]\nGet:3 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 redis-server amd64 5:4.0.9-1ubuntu0.2 [35.4 kB]\nFetched 634 kB in 1s (493 kB/s)\nSelecting previously unselected package libjemalloc1.\r\n(Reading database ... \r(Reading database ... 5%\r(Reading database ... 10%\r(Reading database ... 15%\r(Reading database ... 20%\r(Reading database ... 25%\r(Reading database ... 30%\r(Reading database ... 35%\r(Reading database ... 40%\r(Reading database ... 45%\r(Reading database ... 50%\r(Reading database ... 55%\r(Reading database ... 60%\r(Reading database ... 65%\r(Reading database ... 70%\r(Reading database ... 75%\r(Reading database ... 80%\r(Reading database ... 85%\r(Reading database ... 90%\r(Reading database ... 95%\r(Reading database ... 100%\r(Reading database ... 111540 files and directories currently installed.)\r\nPreparing to unpack .../libjemalloc1_3.6.0-11_amd64.deb ...\r\nUnpacking libjemalloc1 (3.6.0-11) ...\r\nSelecting previously unselected package redis-tools.\r\nPreparing to unpack .../redis-tools_5%3a4.0.9-1ubuntu0.2_amd64.deb ...\r\nUnpacking redis-tools (5:4.0.9-1ubuntu0.2) ...\r\nSelecting previously unselected package redis-server.\r\nPreparing to unpack .../redis-server_5%3a4.0.9-1ubuntu0.2_amd64.deb ...\r\nUnpacking redis-server (5:4.0.9-1ubuntu0.2) ...\r\nSetting up libjemalloc1 (3.6.0-11) ...\r\nSetting up redis-tools (5:4.0.9-1ubuntu0.2) ...\r\nSetting up redis-server (5:4.0.9-1ubuntu0.2) ...\r\nJob for redis-server.service failed because a timeout was exceeded.\r\nSee \"systemctl status redis-server.service\" and \"journalctl -xe\" for details.\r\ninvoke-rc.d: initscript redis-server, action \"start\" failed.\r\n* redis-server.service - Advanced key-value store\r\n   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)\r\n   Active: activating (auto-restart) (Result: timeout) since Sat 2019-12-07 03:41:52 UTC; 32ms ago\r\n     Docs: http://redis.io/documentation,\r\n           man:redis-server(1)\r\n  Process: 24221 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)\r\ndpkg: error processing package redis-server (--configure):\r\n installed redis-server package post-installation script subprocess returned error exit status 1\r\nProcessing triggers for libc-bin (2.27-3ubuntu1) ...\r\nProcessing triggers for systemd (237-3ubuntu10.33) ...\r\nProcessing triggers for man-db (2.8.3-2ubuntu0.1) ...\r\nProcessing triggers for ureadahead (0.100.0-21) ...\r\nErrors were encountered while processing:\r\n redis-server\r\n",
    "stdout_lines": [
        "Reading package lists...",
        "Building dependency tree...",
        "Reading state information...",
        "The following additional packages will be installed:",
        "  libjemalloc1 redis-tools",
        "Suggested packages:",
        "  ruby-redis",
        "The following NEW packages will be installed:",
        "  libjemalloc1 redis-server redis-tools",
        "0 upgraded, 3 newly installed, 0 to remove and 1 not upgraded.",
        "Need to get 634 kB of archives.",
        "After this operation, 3012 kB of additional disk space will be used.",
        "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjemalloc1 amd64 3.6.0-11 [82.4 kB]",
        "Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 redis-tools amd64 5:4.0.9-1ubuntu0.2 [516 kB]",
        "Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 redis-server amd64 5:4.0.9-1ubuntu0.2 [35.4 kB]",
        "Fetched 634 kB in 1s (493 kB/s)",
        "Selecting previously unselected package libjemalloc1.",
        "(Reading database ... ",
        "(Reading database ... 5%",
        "(Reading database ... 10%",
        "(Reading database ... 15%",
        "(Reading database ... 20%",
        "(Reading database ... 25%",
        "(Reading database ... 30%",
        "(Reading database ... 35%",
        "(Reading database ... 40%",
        "(Reading database ... 45%",
        "(Reading database ... 50%",
        "(Reading database ... 55%",
        "(Reading database ... 60%",
        "(Reading database ... 65%",
        "(Reading database ... 70%",
        "(Reading database ... 75%",
        "(Reading database ... 80%",
        "(Reading database ... 85%",
        "(Reading database ... 90%",
        "(Reading database ... 95%",
        "(Reading database ... 100%",
        "(Reading database ... 111540 files and directories currently installed.)",
        "Preparing to unpack .../libjemalloc1_3.6.0-11_amd64.deb ...",
        "Unpacking libjemalloc1 (3.6.0-11) ...",
        "Selecting previously unselected package redis-tools.",
        "Preparing to unpack .../redis-tools_5%3a4.0.9-1ubuntu0.2_amd64.deb ...",
        "Unpacking redis-tools (5:4.0.9-1ubuntu0.2) ...",
        "Selecting previously unselected package redis-server.",
        "Preparing to unpack .../redis-server_5%3a4.0.9-1ubuntu0.2_amd64.deb ...",
        "Unpacking redis-server (5:4.0.9-1ubuntu0.2) ...",
        "Setting up libjemalloc1 (3.6.0-11) ...",
        "Setting up redis-tools (5:4.0.9-1ubuntu0.2) ...",
        "Setting up redis-server (5:4.0.9-1ubuntu0.2) ...",
        "Job for redis-server.service failed because a timeout was exceeded.",
        "See \"systemctl status redis-server.service\" and \"journalctl -xe\" for details.",
        "invoke-rc.d: initscript redis-server, action \"start\" failed.",
        "* redis-server.service - Advanced key-value store",
        "   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)",
        "   Active: activating (auto-restart) (Result: timeout) since Sat 2019-12-07 03:41:52 UTC; 32ms ago",
        "     Docs: http://redis.io/documentation,",
        "           man:redis-server(1)",
        "  Process: 24221 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)",
        "dpkg: error processing package redis-server (--configure):",
        " installed redis-server package post-installation script subprocess returned error exit status 1",
        "Processing triggers for libc-bin (2.27-3ubuntu1) ...",
        "Processing triggers for systemd (237-3ubuntu10.33) ...",
        "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...",
        "Processing triggers for ureadahead (0.100.0-21) ...",
        "Errors were encountered while processing:",
        " redis-server"
    ]
}

RUNNING HANDLER [mariadb : restart mysql] *******************************************************************************************************************************************
task path: /tmp/.bench/playbooks/roles/mariadb/handlers/main.yml:2

PLAY RECAP **************************************************************************************************************************************************************************
localhost                  : ok=47   changed=34   unreachable=0    failed=1    skipped=33   rescued=0    ignored=0

Traceback (most recent call last):
  File "install.py", line 413, in <module>
    install_bench(args)
  File "install.py", line 135, in install_bench
    run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
  File "install.py", line 327, in run_playbook
    success = subprocess.check_call(args, cwd=os.path.join(cwd, 'playbooks'))
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2
```

Full Log: https://privatebin.net/?4665d70d9cc101bb#GmyNBrbvWrAhv4Ux2fWkrQEF32ZvbGXp9zpMTpduWQoj

**Comment by Thunderbottom** [comment]
@yashodhank what does `journalctl -xe` show?

The error literally says `redis-server.service` timed out:

```
Setting up redis-server (5:4.0.9-1ubuntu0.2) ...
Job for redis-server.service failed because a timeout was exceeded.
See \"systemctl status redis-server.service\" and \"journalctl -xe\" for details.
invoke-rc.d: initscript redis-server, action \"start\" failed.
* redis-server.service - Advanced key-value store
   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: timeout) since Sat 2019-12-07 03:41:52 UTC; 32ms ago
     Docs: http://redis.io/documentation,
           man:redis-server(1)
  Process: 24221 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)
dpkg: error processing package redis-server (--configure):
 installed redis-server package post-installation script subprocess returned error exit status 1
```

**Comment by Thunderbottom** [comment]
@yashodhank what does `journalctl -xe` show?

The error literally says `redis-server.service` timed out:

```
Setting up redis-server (5:4.0.9-1ubuntu0.2) ...
Job for redis-server.service failed because a timeout was exceeded.
See "systemctl status redis-server.service" and "journalctl -xe" for details.
invoke-rc.d: initscript redis-server, action "start" failed.
* redis-server.service - Advanced key-value store
   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: timeout) since Sat 2019-12-07 03:41:52 UTC; 32ms ago
     Docs: http://redis.io/documentation,
           man:redis-server(1)
  Process: 24221 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)
dpkg: error processing package redis-server (--configure):
 installed redis-server package post-installation script subprocess returned error exit status 1
```

**Comment by rashedomar** [comment]
same issue

**Comment by Habeebu** [comment]
python3 install.py --production --version 11 --python python2.7 --user magiccraft
Logs are saved under /tmp/logs/easy-install__2020-01-13__09-01.log
Checking System Compatibility...
ubuntu 18 is compatible!
curl already installed!
wget already installed!
git already installed!
pip3 already installed!
WARNING: The directory '/root/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: The directory '/root/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
 [WARNING]: No inventory was parsed, only implicit localhost is available
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
Input MySQL and Frappe Administrator passwords:
Please enter mysql root password:
Re-enter mysql root password:
Please enter the default Administrator user password:
Re-enter Administrator password:
Administrator passwords did not match!
Please enter the default Administrator user password:
Re-enter Administrator password:
Administrator passwords did not match!
Please enter the default Administrator user password:
Re-enter Administrator password:
Passwords saved at ~/passwords.txt
 [WARNING]: No inventory was parsed, only implicit localhost is available
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
 [WARNING]: Could not find aptitude. Using apt-get instead
Traceback (most recent call last):
  File "install.py", line 414, in <module>
    install_bench(args)
  File "install.py", line 230, in install_bench
    run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
  File "install.py", line 351, in run_playbook
    success = subprocess.check_call(args, cwd=os.path.join(cwd, 'playbooks'), stdout=log_stream, stderr=sys.stderr)
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=magiccraft']' returned non-zero exit status 2.

**Comment by atishagrawal** [comment]
@yashodhank @Thunderbottom 
I tried @yashodhank approach of commenting the bind address of the redis server. But I am facing the same issue again. I am using DigitalOcean to host a droplet and Ubuntu 18. Fresh server.
I am posting my error output for your reference.


```
sudo python3 install.py --production --user frappe-user
Logs are saved under /tmp/logs/easy-install__2020-01-23__10-50.log
Checking System Compatibility...
ubuntu 18 is compatible!
curl already installed!
wget already installed!
git already installed!
pip3 already installed!
 [WARNING]: No inventory was parsed, only implicit localhost is available
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
Input MySQL and Frappe Administrator passwords:
Passwords saved at ~/passwords.txt
 [WARNING]: No inventory was parsed, only implicit localhost is available
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
Traceback (most recent call last):
  File "install.py", line 414, in <module>
    install_bench(args)
  File "install.py", line 230, in install_bench
    run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
  File "install.py", line 351, in run_playbook
    success = subprocess.check_call(args, cwd=os.path.join(cwd, 'playbooks'), stdout=log_stream, stderr=sys.stderr)
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe-user']' returned non-zero exit status 2.
```

**Comment by gavindsouza** [comment]
> I tried @yashodhank approach of commenting the bind address of the redis server. But I am facing the same issue again. I am using DigitalOcean to host a droplet and Ubuntu 18. Fresh server.
> I am posting my error output for your reference.

Just ran it on a new DO droplet `Ubuntu 18.04.3 LTS (GNU/Linux 4.15.0-66-generic x86_64)` and worked just fine for me. 

![Screenshot 2020-01-23 at 5 38 29 PM](https://user-images.githubusercontent.com/36654812/72983430-42861c80-3e07-11ea-9619-e43928f02a12.png)
![Screenshot 2020-01-23 at 5 37 38 PM](https://user-images.githubusercontent.com/36654812/72983429-42861c80-3e07-11ea-9672-ff59fc26f6b0.png)

As per the log file `easy-install__2020-01-23__10-30.log` in https://github.com/frappe/bench/issues/905#issuecomment-577625568, it clearly seems like there is a mariadb-client package conflict. Excerpts from your log:
```
    "Reading package lists...",
        "Building dependency tree...",
        "Reading state information...",
        "Some packages could not be installed. This may mean that you have",
        "requested an impossible situation or if you are using the unstable",
        "distribution that some required packages have not yet been created",
        "or been moved out of Incoming.",
        "The following information may help to resolve the situation:",
        "",
        "The following packages have unmet dependencies:",
        " mariadb-client : Depends: mariadb-client-10.2 (>= 1:10.2.30+maria~bionic) but it is not going to be installed"
```
Tinkering with your redis config won't change that.

**Comment by atishagrawal** [comment]
@gavindsouza Just a head's up, I tried the same over a new Droplet again, after your comment and it worked. I don't know what was I doing wrong earlier.

![image](https://user-images.githubusercontent.com/6162950/73057848-779e7780-3eb8-11ea-92c1-7075d201e169.png)

Thanks a lot for your quick replies.

**Comment by creyD** [comment]
I encountered the same issue, reset my vm 3 times now, don't think the issue is on my side. What to do isn't really clear from this, it just seems that it worked one day for you right?

**Comment by weasam** [comment]
> Hi,
> @yashodhank I am facing same issue, I did what you mentioned as solution, no luck still issue showing.
> 
> I am using ubuntu WSL under windows 10.
> 
> ```
> PLAY RECAP *************************************************************************************************************
> localhost                  : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
> 
> Traceback (most recent call last):
>   File "install.py", line 413, in <module>
>     install_bench(args)
>   File "install.py", line 135, in install_bench
>     run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
>   File "install.py", line 327, in run_playbook
>     success = subprocess.check_call(args, cwd=os.path.join(cwd, 'playbooks'))
>   File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2
> ```

Exactly happen to my Ubuntu 18.04.  takes me a lot of time to install ERPNEXT

**Comment by creyD** [comment]
Discussing further in #1091

**Comment by Botanium** [comment]
I am getting the same error on Ubuntu Server 18.04 on a local virtual box. I don't see a proper solution to this error at all.

**Comment by neelbhanushali** [comment]
**$cat /etc/os-release**
```
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

Steps provided [here](https://mariadb.com/kb/en/mariadb-installation-dependancy-problems-and-failed-to-start-service/) solved the problem

**Comment by AliShaatani** [comment]
i tired every fix on this issue and still suffering from that error
`Traceback (most recent call last):
  File "install.py", line 500, in <module>
    install_bench(args)
  File "install.py", line 281, in install_bench
    run_playbook('site.yml', sudo=True, extra_vars=extra_vars)
  File "install.py", line 416, in run_playbook
    success = subprocess.check_call(args, cwd=playbooks_folder, stdout=log_stream, stderr=sys.stderr)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2.
ubuntu@ip-172-31-12-192:~$ File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
subprocess.CalledProcessError: Command '['ansible-playbook', '-c', 'local', 'site.yml', '-vvvv', '-e', '@/tmp/extra_vars.json', '--become', '--become-user=frappe']' returned non-zero exit status 2.
`

**Comment by AliShaatani** [comment]
> **$cat /etc/os-release**
> 
> ```
> NAME="Ubuntu"
> VERSION="20.04.2 LTS (Focal Fossa)"
> ID=ubuntu
> ID_LIKE=debian
> PRETTY_NAME="Ubuntu 20.04.2 LTS"
> VERSION_ID="20.04"
> HOME_URL="https://www.ubuntu.com/"
> SUPPORT_URL="https://help.ubuntu.com/"
> BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
> PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
> VERSION_CODENAME=focal
> UBUNTU_CODENAME=focal
> ```
> 
> Steps provided [here](https://mariadb.com/kb/en/mariadb-installation-dependancy-problems-and-failed-to-start-service/) solved the problem

it doesn't fix the issue

**Comment by dillfrescott** [comment]
Same issue

**Comment by weasam** [comment]
I Have fixed, as remember in my mind is reinstalled by this tutorial, this problem occur could try run and install all python dependencies as requirements, example: pip3 install setuptools
https://www.digitalocean.com/community/tutorials/how-to-install-an-erpnext-stack-on-ubuntu-20-04#step-6-%E2%80%94-installing-frappe-bench-cli
